### PR TITLE
RootLookuper: move Iterate inside RootLookuper to use SingleFlight

### PR DESCRIPTION
**Pull request by amery**
and a 1k buffer

### CODE REVIEW

File: iterator.go [hunk 1]
```diff
@@ -131,13 +157,13 @@ func isIP4(s string) bool {
...
-func hostFromRoot(ctx context.Context, h string) (string, error) {
+func (r RootLookuper) hostFromRoot(ctx context.Context, h string) (string, error) {
 	askRoot := pickRoot()
 	if askRoot == "" {
 		return "", fmt.Errorf("could not pick root")
 	}
 	askRoot = askRoot + ":53"
-	msg, err := Iterate(ctx, h, dns.TypeA, askRoot)
+	msg, err := r.Iterate(ctx, h, dns.TypeA, askRoot)
 	if werr := validateResp(askRoot, msg, err); werr != nil {
```
Comment (amery) [inline iterator.go:167]: is it me of this call to validateResp is redundant? @karasz

**Comment by amery** [comment]
dropped redundant code and fixed handling of r.Exchage()'s error

**Review by karasz — CHANGES_REQUESTED** [review]
Please add this test too:
```go
func TestLookupEdu(t *testing.T) {
z, err := Iterate(context.TODO(), "www.seas.upenn.edu", dns.TypeA, "")
if err != nil {
t.Errorf(err.Error())
}
t.Logf("Edu yeld %s", z.Answer[0].(*dns.A).A.String())
}
```

obviously modified to fit the new situation

**Comment by amery** [comment]
@karasz test added

**Comment by amery** [comment]
@karasz refactored and added the `.edu`

**Review by karasz — APPROVED** [review]
LGTM

**Comment by amery** [comment]
`v0.3.0` tagged

**Pull request merged**